### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by FrankMugagga**
# Acitivities
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
It will return the first user from the database.
Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,7 +9,37 @@ def index
...
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    # title = post_params[:title]
+    # text = post_params[:text]
+    # posts = Post.new(author_id: current_user.id, title: title, text: text, comments_counter: 0, likes_counter: 0 )
```
Comment (Shedrack-Sunday) [inline app/controllers/posts_controller.rb]: - [ ] Kindly do well to remove commented code from your project. This keeps your codebase clustered is done repeatedly wich won't be a good practice 👍

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -9,7 +9,37 @@ def index
...
+    @post = current_user.posts.build(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+
+    if @post.save
+      # redirect_to user_posts_path(current_user, @post), notice: 'Post created successfully'
```
Comment (Shedrack-Sunday) [inline app/controllers/posts_controller.rb]: - [ ] Please do well to remove commented code from your project. This keeps your codebase clustered is done repeatedly wich won't be a good practice

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (Shedrack-Sunday) [inline app/models/user.rb]: - [ ] Kindly do well to remove commented code from your project. This keeps your codebase clustered is done repeatedly wich won't be a good practice 👍

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @FrankMugagga          👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  
- Nice work with the setup✔️  
- Beautiful use of partials ✔️      
 
# Required Changes

- [ ] Kindly take heed to fix the changes suggested below to have a clean code 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Ismailco — APPROVED** [review]
### Hi @FrankMugagga, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**